src/com/TTT/GameOptionsUI.java

```diff
@@ -0,0 +1,43 @@
+package com.TTT;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+
+public class GameOptionsUI {
+    private PrintStream out;
+    private InputStream in;
+    private Scanner scanner;
+
+    public GameOptionsUI(PrintStream out, InputStream in) {
+        this.out = out;
+        this.in = in;
+        scanner = new Scanner(in);
+        scanner.useDelimiter("\n");
+    }
+
+    public int boardSize() {
+        out.print("Would you like a 3x3 or 4x4 board? Please enter 3 or 4: ");
+
+        String selection = getBoardSizeSelection();
+        while(!validSelection(selection) || ((Integer.parseInt(selection) != 3) && (Integer.parseInt(selection) != 4))) {
```

Comment:
You could wrap these three method calls into one so the while loop reads nicer `while(invalidInput(selection)`